src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Day/DiaryDayNotesInfoViewController.swift

```diff
@@ -0,0 +1,93 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+
+class DiaryDayNotesInfoViewController: DynamicTableViewController, DismissHandling {
+
+	// MARK: - Attributes
+
+	var finished: (() -> Void)?
+
+	// MARK: - Initializers
+	
+	init() {
+		self.viewModel = DiaryDayNotesInfoViewModel()
+		super.init(nibName: nil, bundle: nil)
+	}
+
+	@available(*, unavailable)
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	// MARK: - View Lifecycle Methods
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		setupTableView()
+		setupRightBarButtonItem()
+	}
+
+	// MARK: - Overrides
+
+	override var navigationItem: UINavigationItem {
+		navigationFooterItem
+	}
+
+	private lazy var navigationFooterItem: ENANavigationFooterItem = {
+		let item = ENANavigationFooterItem()
+
+		item.primaryButtonTitle = AppStrings.NewVersionFeatures.buttonContinue
+		item.isPrimaryButtonEnabled = true
+		item.isSecondaryButtonHidden = true
+
+		return item
+	}()
+	
+	// MARK: - Protocol DismissHandling
+	
+	func wasAttemptedToBeDismissed() {
+		finished?()
+	}
+	
+	// MARK: - Protocol ENANavigationControllerWithFooterChild
+
+	func navigationController(_ navigationController: ENANavigationControllerWithFooter, didTapPrimaryButton button: UIButton) {
+		finished?()
+	}
+
+	// MARK: - Private API
+
+	private let viewModel: DiaryDayNotesInfoViewModel
+
+	private func setupRightBarButtonItem() {
+		navigationItem.rightBarButtonItem = CloseBarButtonItem(
+			onTap: { [weak self] in
+				self?.finished?()
+			}
+		)
+	}
```

Comment:
we have a new logic for the closeButton - please have look inside the DismissHandling protocol.
just add ```navigationItem.rightBarButtonItem = dismissHandlingCloseBarButton``` inside the ```init()``` method after ```super.init()``` and you are done

Reply:
Adjusted according to DidmissHandling 👍